EIPS/eip-4844.md

```diff
@@ -307,45 +343,16 @@ def get_intrinsic_gas(tx: SignedBlobTransaction, pre_state: ExecState) -> int:
     intrinsic_gas += len(tx.message.blob_versioned_hashes) * get_blob_gas(pre_state)
     return intrinsic_gas
 
-def get_blob_gas(pre_state: ExecState) -> int:
-    blocks_since_start = get_current_block(pre_state) - FORK_BLKNUM
-    targeted_total = blocks_since_start * TARGET_BLOB_TXS_PER_BLOCK
-    actual_total = read_storage(
-        pre_state,
-        SYSTEM_STATE_ADDRESS,
-        TOTAL_BLOB_TXS_STORAGE_SLOT
-    )
-    if actual_total < targeted_total:
+def get_blob_gas(header: Header) -> int:
+    if header.blobs == 0:
         return 0
     else:
         return fake_exponential(
-            actual_total - targeted_total,
+            header.blobs,
             GASPRICE_UPDATE_FRACTION_PER_BLOB
         )
 ```
 
-We update at the end of a block, as follows:
```

Comment:
We might need something equivalent for setting the new excess blobs header field, using `calc_excess_blobs`. I.e. specify how to calculate the total number of blobs in the block by iterating through all blob txs and summing up their blobs counts.

Reply:
I think this is covered by the link "The value of `excess_blobs` can be calculated ... ".

Reply:
I think we need to specify where the input value `new_blobs` to the `calc_excess_blobs` function comes from though.

Reply:
"... using the parent header and number of blobs in the block."

Reply:
;)